### PR TITLE
adds access token scope configuration API for AppSSO AuthServer CR

### PR DESCRIPTION
**Pull request by ddubson**
# Which other branches do you want a technical writer to cherry-pick this PR to (if any)?

It's best to PR to the most recent relevant branch and leave all the cherry-picking to the
writer, except where earlier branches require factual changes to the content.
For more information about the branches, see https://github.com/pivotal/docs-tap#branches

### CODE REVIEW

File: app-sso/service-operators/configure-authorization.hbs.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+An application or `Workload` may protect certain resources based on user's level of authorization. Within OAuth 2, the application
+with protected resources (_the Resource Server_) verifies if the access token provided contains the scope(s) to perform
+an action on a given protected resource.
+
+_Example_: excerpt from a Spring Boot application that is an OAuth2 Resource Server, protecting its _message_ 
+API endpoints `/message/**`:
+
+```java
+http.authorizeExchange(exchanges -> exchanges
+			.pathMatchers("/message/**").hasAuthority("SCOPE_message:read")
```
Comment (fhanik) [inline app-sso/service-operators/configure-authorization.hbs.md:15]: ```SCOPE_message:read```

did we switch away from period `.` to colon `:` or is spring boot doing some translation here?

File: app-sso/service-operators/configure-authorization.hbs.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+# ...
+spec:
+  identityProviders:
+    - name: my-openid-provider
+      openid:
+        rolesToScopes:
+          - fromRole: "hr"           # -> role "hr" is mapped to "hr.read", "hr.write" scopes
+            toScopes:                #    only users with "hr" role may be issued access token with these scopes
+              - "hr.read"            # ^^
+              - "hr.write"           # ^^
```
Comment (fhanik) [inline app-sso/service-operators/configure-authorization.hbs.md:66]: I could be reading this wrong, but shouldn't there be an `accessToken` key in this structure?

File: app-sso/service-operators/configure-authorization.hbs.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+              toScopes:                #    only users with "hr" role may be issued access token with these scopes
+                - "hr.read"            # ^^
+                - "hr.write"           # ^^
+            - fromRole: "finance"      # -> role "finance" is mapped to "finance" scope
+              toScopes:                #    only users with "finance" role may be issued an access token with this scope
+                - "finance"            # ^^
+            - fromRole: "marketing"    # -> role "marketing" is mapped to "marketing-reader", "marketing-writer" scopes
+              toScopes:                #    only users with "marketing" role may be issued an access token with these scopes
+                - "marketing-reader"   # ^^
+                - "marketing-writer"   # ^^
```
Comment (fhanik) [inline app-sso/service-operators/configure-authorization.hbs.md:157]: in this were mixing naming conventions of scopes. I am guessing that are we're doing that intentionally to demonstrate that scope names are just arbitrary strings?

1. `developer.read` - namespaced by dot `.`
2. `finance` - no topology
3. `marketing-reader` - namespaced by dash `-`

File: release-notes.hbs.md [hunk 4]
```diff
@@ -12,8 +12,11 @@ This topic contains release notes for Tanzu Application Platform v1.5.
 
 ### <a id='1-5-0-appsso-new-features'></a> Application Single Sign-On (AppSSO)
 
-- Adds a consistent roles claim mapping API across OpenID, LDAP, and SAML identity providers.
-- Introduces roles claim filtering API in an `AuthServer`s identity provider configuration.
+- Adds a consistent roles claim mapping API across OpenID, LDAP, and SAML identity providers in `AuthServer` resource.
```
Comment (fhanik) [inline release-notes.hbs.md:15]: Role mappings to accessToken, idToken or both? May be helpful to clarify the scope of this change. (pun intended)

**Review by fhanik — COMMENTED** [review]
Looking great, just minor comments and questions

File: app-sso/service-operators/configure-authorization.hbs.md (hunk 2 above)
Comment (ddubson) [reply app-sso/service-operators/configure-authorization.hbs.md:66]: `accessToken` is at the root of this subtree for an IdP:

```
accessToken:
   scope:
      defaults: []
      rolesToScopes: []
```

Comment (ddubson) [reply app-sso/service-operators/configure-authorization.hbs.md:66]: Ahh oops! 😶‍🌫️ I missed it, good catch! Will add.

File: release-notes.hbs.md (hunk 4 above)
Comment (ddubson) [reply release-notes.hbs.md:15]: Will clarify this point.

**Review by candicenonsense — APPROVED** [review]
This looks good.  Filip got to my questions already. :)

File: app-sso/service-operators/configure-authorization.hbs.md (hunk 1 above)
Comment (fhanik) [reply app-sso/service-operators/configure-authorization.hbs.md:15]: I'm ok with either format. My only concern was that if we use `message.read` anywhere else in our document, there is a risk that the reader believes that `message.read` must be converted to `message:read` in a Boot app.

**Review by fhanik — APPROVED** [review]

**Pull request closed without merging**